backend/app/services/execution/dispatcher.py

```
@@ -927,9 +1054,45 @@ async def cancel(
             return await self._cancel_websocket(request, target)
         elif target.mode == CommunicationMode.POLLING:
             return await self._cancel_sse(request, target)
+        elif target.mode == CommunicationMode.INPROCESS:
+            return await self._cancel_inprocess(request, target)
         else:
             return await self._cancel_http(request, target)
 
+    async def _cancel_inprocess(
+        self,
+        request: ExecutionRequest,
+        target: ExecutionTarget,
```

Ruff (0.15.4)

[warning] Unused method argument: `target`

(ARG002)